pkg/kubectl/cmd/top_node_test.go

```diff
@@ -242,3 +263,164 @@ func TestTopNodeWithLabelSelectorMetrics(t *testing.T) {
 		}
 	}
 }
+
+func TestTopNodeAllMetricsFromMetricsServer(t *testing.T) {
+	initTestErrorHandler(t)
+	expectedMetrics, nodes := testNodeV1beta1MetricsData()
+	expectedNodePath := fmt.Sprintf("/%s/%s/nodes", apiPrefix, apiVersion)
+
+	f, tf, codec, ns := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.Client = &fake.RESTClient{
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
```

Comment:
if this is for discovery, this should be structured so that you can use a fake version of the discovery client to accomplish this.  You should basically never have to use a fake HTTP client except for when simulating a Heapster response

Reply:
It is, but the problem is that the underlying testFactory deeply integrates only with the fake rest client. My feeling is that changing that is out of scope for this PR, it would have a lot of impact on the kubectl code, and those changes would be much larger then the PR so far.

Reply:
ack, we'll put that off until later.